Sources/JS/helpers.js

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+var maxCollectedProperties = 200
+var possibleGroups = [
+    {name: "Symbol", belongsToGroup: function(obj){typeof obj == 'symbol'}},
+    {name: "String", belongsToGroup: function(obj){return obj instanceof String}},
+    {name: "RegExp", belongsToGroup: function(obj){return obj instanceof RegExp}},
+    {name: "Array", belongsToGroup: function(obj){return obj instanceof Array}},
+    {name: "Map", belongsToGroup: function(obj){return obj instanceof Map}},
+    {name: "Promise", belongsToGroup: function(obj){return obj instanceof Promise}},
+    {name: "WeakMap", belongsToGroup: function(obj){return obj instanceof WeakMap}},
+    {name: "Set", belongsToGroup: function(obj){return obj instanceof Set}},
+    {name: "WeakSet", belongsToGroup: function(obj){return obj instanceof WeakSet}},
+    {name: "ArrayBuffer", belongsToGroup: function(obj){return obj instanceof ArrayBuffer}},
+    {name: "DataView", belongsToGroup: function(obj){return obj instanceof DataView}},
+    {name: "Uint8Array", belongsToGroup: function(obj){return obj instanceof Uint8Array}},
+    {name: "Int8Array", belongsToGroup: function(obj){return obj instanceof Int8Array}},
+    {name: "Uint16Array", belongsToGroup: function(obj){return obj instanceof Uint16Array}},
+    {name: "Int16Array", belongsToGroup: function(obj){return obj instanceof Int16Array}},
+    {name: "Uint32Array", belongsToGroup: function(obj){return obj instanceof Uint32Array}},
+    {name: "Int32Array", belongsToGroup: function(obj){return obj instanceof Int32Array}},
+    {name: "Float32Array", belongsToGroup: function(obj){return obj instanceof Float32Array}},
+    {name: "Float64Array", belongsToGroup: function(obj){return obj instanceof Float64Array}},
+    {name: "Uint8ClampedArray", belongsToGroup: function(obj){return obj instanceof String}},
+    {name: "Object", belongsToGroup: function(obj){return obj instanceof Object}},
```

Comment:
Maybe make a comment here that this must be the last entry since it will also match for pretty much all other types?